### PR TITLE
Support usage of ** inside a pathexp being parsed

### PR DESCRIPTION
**Pull request by jeffandersen**
- Adds support for path inputs to contain `**`
- All paths will still be stored/used internally as their complete equivalent

### CODE REVIEW

**Comment by jbowes** [comment]
you'll need to update the grammar def in the comment at the top of `pathexp.go`

File: pathexp/pathexp.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ var (
 	slug           = regexp.MustCompile(`^` + slugstr + `$`)
 	globRe         = regexp.MustCompile(`^(` + slugstr + `)(\*?)$`)
 	fullglobOrGlob = regexp.MustCompile(`(^\*$)|(?:^(` + slugstr + `)(\*?)$)`)
+	doubleGlob     = regexp.MustCompile(`^\*\*$`)
```
Comment (jbowes) [inline pathexp/pathexp.go]: wanna add some tests for doubleglob?

File: pathexp/pathexp.go [hunk 2]
```diff
@@ -283,27 +284,51 @@ func New(org, project string, envs, services, identities, instances []string) (*
...
 		return nil, errors.New("path expressions must start with '/'")
 	}
+
 	// remove leading empty section
 	parts = parts[1:]
 
+	// identify double glob and replace remaining segments with glob
+	hasDoubleGlob := false
+	segments := make([]string, 6)
+	for i := range segments {
```
Comment (jbowes) [inline pathexp/pathexp.go]: double splatting needs to be greedy:`/jbowes/foo/**/default/foo` -> `/jbowes/foo/*/default/*/*/foo` (but should be `/jbowes/foo/*/*/*/default/foo`).

Comment (jbowes) [inline pathexp/pathexp.go]: two double splats should be invalid, as they'd be ambiguous:
`/jbowes/foo/**/default/**/foo` could be either:
- `/jbowes/foo/*/*/default/*/foo`
- `/jbowes/foo/*/default/*/*/foo`

**Comment by jeffandersen** [comment]
Gonna roll this into some `torus ls` improvements, changing this opened a can of worms.

**Pull request closed without merging**